powersimdata/scenario/execute.py

```diff
@@ -203,9 +207,30 @@ def __init__(self, data_access, scenario_info, grid, ct):
 
         self.REL_TMP_DIR = self._data_access.tmp_folder(self.scenario_id)
 
+    def _adjust_pmin(self):
+        mi = self.grid.model_immutables
+        plant = self.grid.plant
+        pmin_factor = mi.plants["pmin_as_share_of_pmax"]
+
+        def _scale(x):
+            factor = pmin_factor.get(x.type)
+            return x.Pmin if factor is None else factor * x.Pmax
+
+        plant.Pmin = plant.apply(_scale, axis=1)
+
+        # set Pmin to 0 for generators that are off or profile based
+        profile_resource = list(mi.plants["profile_resources"])
+        plant.loc[plant.type.isin(profile_resource), "Pmin"] = 0
+        plant.loc[plant.status == 0, "Pmin"] = 0
```

Comment:
Maybe out of the scope of this PR, two high level questions:
1. `hydro` is one of `profile_resources`, the `Pmin` of which will be set 0 at the end, however the `pmin_factor` of `hydro` is grid dependent .
2. We will need to build user interface to pass user specified `pmin_as_share_of_pmax` (either plant based or plant type based or plant type * zone based) in the future, right?

Reply:
1. True. Just pointing out such entries are modified twice in the current process.
2. Agreed.